Sources/IO/Geometry/GLTFImporter/Reader.js

```diff
@@ -0,0 +1,609 @@
+      primitive.extensions.KHR_draco_mesh_compression
+    );
+  }
+
+  const polyData = vtkPolyData.newInstance();
+  const cells = vtkCellArray.newInstance();
+  const pointData = polyData.getPointData();
+
+  const attrs = Object.entries(primitive.attributes);
+  attrs.forEach(async ([attributeName, accessor]) => {
+    switch (attributeName) {
+      case SEMANTIC_ATTRIBUTE_MAP.POSITION: {
+        const position = primitive.attributes.position.value;
+        polyData
+          .getPoints()
+          .setData(position, primitive.attributes.position.component);
+        break;
+      }
+      case SEMANTIC_ATTRIBUTE_MAP.NORMAL: {
+        const normals = primitive.attributes.normal.value;
+        pointData.setNormals(
+          vtkDataArray.newInstance({
+            name: 'Normals',
+            values: normals,
+            numberOfComponents: primitive.attributes.normal.components,
+          })
+        );
+        break;
+      }
+      case SEMANTIC_ATTRIBUTE_MAP.COLOR_0: {
+        const color = primitive.attributes.color.value;
+        pointData.setScalars(
+          vtkDataArray.newInstance({
+            name: 'Scalars',
+            values: color,
+            numberOfComponents: primitive.attributes.color.components,
+          })
+        );
+        break;
+      }
+      case SEMANTIC_ATTRIBUTE_MAP.TEXCOORD_0: {
+        const tcoords0 = primitive.attributes.texcoord0.value;
+        const da = vtkDataArray.newInstance({
+          name: 'TEXCOORD_0',
+          values: tcoords0,
+          numberOfComponents: primitive.attributes.texcoord0.components,
+        });
+        pointData.addArray(da);
+        pointData.setActiveTCoords(da.getName());
+        break;
+      }
+      case SEMANTIC_ATTRIBUTE_MAP.TEXCOORD_1: {
+        const tcoords = primitive.attributes.texcoord1.value;
+        const dac = vtkDataArray.newInstance({
+          name: 'TEXCOORD_1',
+          values: tcoords,
+          numberOfComponents: primitive.attributes.texcoord1.components,
+        });
+        pointData.addArray(dac);
+        break;
+      }
+      case SEMANTIC_ATTRIBUTE_MAP.TANGENT: {
+        const tangent = primitive.attributes.tangent.value;
+        const dat = vtkDataArray.newInstance({
+          name: 'Tangents',
+          values: tangent,
+          numberOfComponents: primitive.attributes.tangent.components,
+        });
+        pointData.addArray(dat);
+        break;
+      }
+      default:
+        vtkWarningMacro(`Unhandled attribute: ${attributeName}`);
+    }
+  });
+
+  // Handle indices if available
+  if (primitive.indices !== undefined) {
+    const indices = primitive.indices.value;
+    const nCells = indices.length - 2;
+
+    switch (mode) {
+      case MODES.GL_LINE_STRIP:
+      case MODES.GL_TRIANGLE_STRIP:
+      case MODES.GL_LINE_LOOP:
+        vtkWarningMacro('GL_LINE_LOOP not implemented');
+        break;
+      default:
+        for (let cellId = 0; cellId < nCells; cellId += 3) {
```

Comment:
you could call `cells.resize(4 * indices.length / 3)` to limit the number of memory reallocations when call `insertNextCell()` hence increase performance